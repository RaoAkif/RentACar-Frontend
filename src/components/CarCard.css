.rent-card {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.rental-carousel-container {
  width: 100%;
}

.car-image-div {
  width: 14rem;
  min-height: 20rem;
  display: flex;
  align-items: center;
  justify-content: center;
}

.car-image {
  width: 100%;
}

.rent-car-info {
  margin-bottom: 4rem;
}

.rent-car-info h3 {
  font-size: 3rem;
  text-align: center;
  margin-bottom: 4rem;
}

.rent-table {
  width: 20rem;
  font-size: 1.6rem;
}

.rent-table span {
  font-weight: bold;
}

@media only screen and (min-width: 480px) {
  .rental-carousel-container {
    width: 70%;
  }

  .rent-car-info h3 {
    font-size: 1.8rem;
  }

  .rent-table {
    width: 14rem;
    font-size: 1.2rem;
  }
}
