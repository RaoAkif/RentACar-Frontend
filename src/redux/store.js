import { configureStore } from '@reduxjs/toolkit';
import cakeReducer from './cake/cakeSlice';
import icecreamReducer from './icecream/icecreamSlice';
import userReducer from './user/userSlice';

const store = configureStore({
  reducer: {
    cake: cakeReducer,
    icecream: icecreamReducer,
    user: userReducer,
  },
});

export default store;
