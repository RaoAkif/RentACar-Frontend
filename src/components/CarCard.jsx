import React from 'react';
import PropTypes from 'prop-types';
import './CarCard.css';

function RentCard(props) {
  const {
    carName,
    model,
    price,
    image,
    city,
    date,
  } = props;

  return (
    <div className="rent-card">
      <div className="car-image-div">
        <img className="car-image" src={image} alt={carName} />
      </div>
      <div className="rent-card-info">
        <div className="name-model">
          <h3>{carName}</h3>
          <p>{model}</p>
        </div>
        <p>{price}</p>
        <p>{city}</p>
        <p>{date}</p>
      </div>
    </div>
  );
}

RentCard.defaultProps = {
  city: '',
  date: '',
};

RentCard.propTypes = {
  carName: PropTypes.string.isRequired,
  model: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  city: PropTypes.string,
  date: PropTypes.string,
};

export default RentCard;
