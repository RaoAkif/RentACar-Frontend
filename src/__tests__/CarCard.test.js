import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import store from '../redux/configureStore';
import CarCard from '../components/CarCard';

describe("CarCard", () => {
  const car = { id: 1, name: 'Audi', model: 'R3', price: '100', image: 'https://www.audi.com/content/dam/audi/us/en/all-models/a4/a4-allroad-quattro/2020/models-a4-allroad-quattro-2020-hero.jpg/_jcr_content/renditions/cq5dam.web.1200.675.jpeg' };

  it("should render correctly", () => {
    const { container } = render(
      <BrowserRouter>
        <Provider store={store}>
          <CarCard
            carName = {car.name}
            model = {car.model}
            price = {car.price}
            image = {car.image}
          />
        </Provider>
        ,
      </BrowserRouter>,
    );

    expect(container).toMatchSnapshot();
  });
});
