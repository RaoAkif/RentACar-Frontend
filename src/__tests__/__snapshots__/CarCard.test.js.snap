// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarCard should render correctly 1`] = `
<div>
  <div
    class="rent-card"
  >
    <div
      class="car-image-div"
    >
      <img
        class="car-image"
      />
    </div>
    <div
      class="rent-car-info"
    >
      <h3 />
      <table
        class="table table-striped table-bordered rent-table"
      >
        <tbody>
          <tr>
            <td
              class="text-start"
            >
              <span>
                Model:
              </span>
            </td>
            <td
              class="text-end"
            />
          </tr>
          <tr>
            <td
              class="text-start"
            >
              <span>
                Rent:
              </span>
            </td>
            <td
              class="text-end"
            />
          </tr>
          <tr>
            <td
              class="text-start"
            >
              <span>
                City:
              </span>
            </td>
            <td
              class="text-end"
            />
          </tr>
          <tr>
            <td
              class="text-start"
            >
              <span>
                Date:
              </span>
            </td>
            <td
              class="text-end"
            />
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  ,
</div>
`;
